libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt

```diff
@@ -36,55 +61,59 @@ object MapboxMetricsReporter : MetricsReporter {
         accessToken: String,
         userAgent: String
     ) {
-        mapboxTelemetry = MapboxTelemetry(context, accessToken, userAgent)
-        mapboxTelemetry.enable()
-    }
-
-    // For test purposes only
-    internal fun init(
-        mapboxTelemetry: MapboxTelemetry,
-        jobControlFactory: InternalJobControlFactory,
-    ) {
-        this.mapboxTelemetry = mapboxTelemetry
-        ioJobController = jobControlFactory.createIOScopeJobControl()
-        mapboxTelemetry.enable()
+        val eventsServerOptions = EventsServerOptions(accessToken, userAgent, null)
+        eventsService = EventsServiceProvider.provideEventsService(eventsServerOptions)
+        telemetryService = TelemetryServiceProvider.provideTelemetryService(eventsServerOptions)
+        eventsService.registerObserver(eventsServiceObserver)
     }
 
     /**
      * Toggle whether or not you'd like to log [mapboxTelemetry] events.
      *
      * @param isDebugLoggingEnabled true to enable logging, false to disable logging
      */
+    @Deprecated("toggleLogging is deprecated and do no operations")
     @JvmStatic
     fun toggleLogging(isDebugLoggingEnabled: Boolean) {
-        mapboxTelemetry.updateDebugLoggingEnabled(isDebugLoggingEnabled)
+        // do nothing
     }
 
     /**
-     * Disables metrics reporting and ends [mapboxTelemetry] session.
+     * Disables metrics reporting and ends [EventsServiceInterface] and [TelemetryService] session.
      * This method also removes metrics observer and stops background thread used for
      * events dispatching.
      */
     @JvmStatic
     fun disable() {
         removeObserver()
-        mapboxTelemetry.disable()
+        eventsService.unregisterObserver(eventsServiceObserver)
         ioJobController.job.cancelChildren()
     }
 
     /**
      * Adds an event to the metrics reporter when this event occurs.
      */
     override fun addEvent(metricEvent: MetricEvent) {
-        metricEvent.toTelemetryEvent()?.let {
-            mapboxTelemetry.push(it)
+        eventsService.sendEvent(
+            Event(EventPriority.IMMEDIATE, metricEvent.toValue(), null)
```

Comment:
Does every event need to be sent with immediate priority? Don't we want to do batching as with the old implementation?

Reply:
Do you mean `QUEUED`? It was added by the core team, suppose for tests. @tatiana-yan could you provide what is recommended `Priority` for events?

Reply:
after the sync with @tarigo: for single events (not stream) recommended using `IMMEDIATE` priority to not lose any. Any of our events are single, so we should be good to keep it as is.